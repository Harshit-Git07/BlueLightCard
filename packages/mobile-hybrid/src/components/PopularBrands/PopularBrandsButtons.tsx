import React from 'react';
import { Brand } from './types';
import Image from '@/components/Image/Image';

interface Props {
  brands: Brand[];
  limit?: number;
  imageWrapperClass: string;
  imageClass: string;
  onButtonClick?: (data: any) => void;
}

const PopularBrandsButtons: React.FC<Props> = ({
  brands,
  limit,
  imageWrapperClass,
  imageClass,
  onButtonClick,
}: Props) => {
  return (
    <>
      {brands.map((brand: any, index: any) => {
        if (limit && index > limit) return; // limit to 8 items

        return (
          <button
            key={brand.id}
            className={imageWrapperClass}
            onClick={() => onButtonClick?.(brand.id)}
            aria-label={brand.brandName}
          >
            <Image src={brand.imageSrc} alt={brand.brandName} className={imageClass} />
          </button>
        );
      })}
    </>
  );
};

export default PopularBrandsButtons;
