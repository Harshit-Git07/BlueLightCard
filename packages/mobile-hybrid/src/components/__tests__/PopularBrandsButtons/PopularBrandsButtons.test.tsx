import React from 'react';
import { render, screen, within, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PopularBrandsButtons from '@/components/PopularBrands/PopularBrandsButtons';

const brands = [
  {
    id: 22,
    imageSrc: '/images/image.jpg',
    brandName: 'Gym King',
  },
  {
    id: 23,
    imageSrc: '/images/image.jpg',
    brandName: 'M&S',
  },
  {
    id: 24,
    imageSrc: '/images/image.jpg',
    brandName: 'Asda',
  },
  {
    id: 25,
    imageSrc: '/images/image.jpg',
    brandName: 'Burger King',
  },
  {
    id: 26,
    imageSrc: '/images/image.jpg',
    brandName: 'Toyota',
  },
  {
    id: 27,
    imageSrc: '/images/image.jpg',
    brandName: 'McDonalds',
  },
  {
    id: 28,
    imageSrc: '/images/image.jpg',
    brandName: 'Superdry',
  },
  {
    id: 29,
    imageSrc: '/images/image.jpg',
    brandName: 'Sainsburys',
  },
  {
    id: 30,
    imageSrc: '/images/image.jpg',
    brandName: 'David Lloyd',
  },
];

// Mock the next/image component
jest.mock('next/image', () => {
  return function MockImage(props: any) {
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    return <img {...props} />;
  };
});

jest.useFakeTimers(); // Use fake timers for testing

describe('PopularBrandsButtons', () => {
  const mockProps = {
    brands: brands,
    imageWrapperClass: 'imageWrapperClass',
    imageClass: 'imageClass',
  };

  it('renders the component', () => {
    const { container } = render(<PopularBrandsButtons {...mockProps} />);

    window.scrollY = 100;
    window.dispatchEvent(new Event('scroll'));

    jest.advanceTimersByTime(750);

    expect(container).toMatchSnapshot();
  });

  it('calls onClick when button is clicked', () => {
    const handleClick = jest.fn(); // Create a mock function
    render(
      <PopularBrandsButtons {...mockProps} onButtonClick={() => handleClick('Test message')} />,
    );

    const button = screen.getByLabelText(/Gym King/i);
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1); // Check if the mock function was called once
  });

  it('renders buttons with images', () => {
    const { getByAltText } = render(<PopularBrandsButtons {...mockProps} />);

    brands.forEach((brand) => {
      const image = getByAltText(brand.brandName) as HTMLImageElement;

      expect(image).toBeInTheDocument();
      expect(image.src).toContain('/images/image.jpg');
    });
  });

  it('renders when it has at least 8 items', () => {
    const { container } = render(<PopularBrandsButtons {...mockProps} limit={7} />);
    const buttons = container.querySelectorAll(
      '.imageWrapperClass',
    ) as unknown as HTMLButtonElement[];

    expect(buttons[0]).toBeInTheDocument();

    expect(buttons).toHaveLength(8);
  });
});
