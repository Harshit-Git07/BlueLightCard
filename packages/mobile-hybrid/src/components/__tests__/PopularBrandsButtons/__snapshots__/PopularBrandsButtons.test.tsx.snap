// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularBrandsButtons renders the component 1`] = `
<div>
  <button
    aria-label="Gym King"
    class="imageWrapperClass"
  >
    <img
      alt="Gym King"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="M&S"
    class="imageWrapperClass"
  >
    <img
      alt="M&S"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="Asda"
    class="imageWrapperClass"
  >
    <img
      alt="Asda"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="Burger King"
    class="imageWrapperClass"
  >
    <img
      alt="Burger King"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="Toyota"
    class="imageWrapperClass"
  >
    <img
      alt="Toyota"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="McDonalds"
    class="imageWrapperClass"
  >
    <img
      alt="McDonalds"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="Superdry"
    class="imageWrapperClass"
  >
    <img
      alt="Superdry"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="Sainsburys"
    class="imageWrapperClass"
  >
    <img
      alt="Sainsburys"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
  <button
    aria-label="David Lloyd"
    class="imageWrapperClass"
  >
    <img
      alt="David Lloyd"
      blurdataurl="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=="
      class="imageClass"
      placeholder="blur"
      src="/images/image.jpg"
    />
  </button>
</div>
`;
